python/llm/src/bigdl/llm/transformers/models/llama.py

```diff
@@ -265,7 +265,8 @@ def llama_attention_forward_4_31(
     value_states = repeat_kv(value_states, self.num_key_value_groups).to(device,
                                                                          dtype=attention_dtype)
 
-    if fsdp_flag:
+    if fsdp_flag and query_states.shape[2] == key_states.shape[2]:
```

Comment:
add this check to `use_flash_attention`?

Reply:
In our current code, we put `use_flash_attention` at the beginning of `llama_attention_forward_4_31`, usage likes `use_flash_attention(hidden_states)`, which means at this time, it only has hidden_states, don't know query_states / key_states .
If we want to put this check into `use_flash_attention`, maybe we can acheive it by delaying the check to line 261.
Do you prefer this ?

Reply:
Have refactored this part, would you mind taking a look again?